src/pybind/mgr/dashboard/frontend/src/app/ceph/dashboard/osd-data.pipe.ts

```diff
@@ -0,0 +1,57 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+import _ from 'lodash';
+
+@Pipe({
+  name: 'osdDataPipe'
+})
+export class OsdDataPipe implements PipeTransform {
+  transform(value: any): any {
+    if (!value) {
+      return '';
+    }
+
+    let nearFullcount = 0;
+    let fullCount = 0;
+
+    _.each(value, (osd) => {
+      if (
+        Math.ceil((osd.stats.stat_bytes_used / osd.stats.stat_bytes) * 100) > 90 &&
+        Math.ceil((osd.stats.stat_bytes_used / osd.stats.stat_bytes) * 100) !== 100
+      ) {
+        nearFullcount++;
+      }
+      if (Math.ceil((osd.stats.stat_bytes_used / osd.stats.stat_bytes) * 100) === 100) {
+        fullCount++;
+      }
+    });
```

Comment:
There's no need to perform this calculation. As soon as OSDs reach the nearfull/full ratios, the ceph [health will start reporting it](https://docs.ceph.com/en/latest/rados/troubleshooting/troubleshooting-osd/#no-free-drive-space):
`# ceph health --format=json | jq`
```json
{
  "status": "HEALTH_WARN",
  "checks": {
    "OSD_NEARFULL": {
      "severity": "HEALTH_WARN",
      "summary": {
        "message": "3 nearfull osd(s)",
        "count": 3
      },
      "muted": false
    },
    "POOL_NEARFULL": {
      "severity": "HEALTH_WARN",
      "summary": {
        "message": "2 pool(s) nearfull",
        "count": 2
      },
      "muted": false
    }
  },
  "mutes": []
}
```
You can also get this info from the osd_map (probably the best place).

Reply:
Done.